src/gnetcli_adapter/gnetcli_adapter.py

```diff
@@ -396,84 +432,133 @@ async def deploy(self, device: Device, cmds: CommandList, args: DeployOptions, p
+                await self.api.upload(hostname=device.fqdn, files=files, host_params=host_params)
+                tracker.files_uploaded()
+
+            async with self.api.cmd_session(hostname=device.fqdn) as sess:
+                try:
+                    seen_exc, res = await self._deploy_cmds(
+                        host_params=host_params,
+                        sess=sess,
+                        run_cmds=run_cmds,
+                        tracker=tracker,
+                    )
+                    if not seen_exc and do_reload:
+                        reload_ecx, reload_res = await self._reload_cmds(
+                            host_params=host_params,
+                            sess=sess,
+                            reload_cmds=reload_cmds,
+                            tracker=tracker,
+                        )
+                        seen_exc.extend(reload_ecx)
+                        res.extend(reload_res)
+                except Exception as e:
+                    trace = traceback.format_exc()
+                    tracker.command_done_error(trace)
+                    seen_exc = [e]
+                    res = []
+
+            if seen_exc:
+                tracker.finish_err(f"Seen {len(seen_exc)} exceptions")
+            else:
+                tracker.finish_ok("All done")
+            return seen_exc, res
+
+    async def _deploy_cmds(
+            self,
+            host_params: HostParams,
+            sess: GnetcliSessionCmd,
+            run_cmds: CommandList,
+            tracker: ProgressTracker,
+    ) -> tuple[list[Exception], list[pb.CMDResult]]:
+        if not run_cmds:
+            return [], []
+        tracker.start_group("Running commands")
+        seen_exc: list[Exception] = []
+        result: List[pb.CMDResult] = []
+        for cmd in run_cmds:
+            tracker.run_command(cmd.cmd)
+            try:
+                res = await sess.cmd(
+                    cmd=cmd.cmd,
+                    cmd_timeout=cmd.timeout,
+                    host_params=host_params,
+                    qa=parse_annet_qa(cmd.questions or []),
+                    trace=True,
+                )
+            except EOFError as e:
+                if cmd.suppress_eof:
+                    tracker.command_done_error("Suppressed EOF")
+                    break  # we can't exec subsequent cmds
+                seen_exc.append(e)
+                return seen_exc, result
+            if res.status == 0:
+                tracker.command_done_ok(res)
+                result.append(res)
+            else:
+                if cmd.suppress_nonzero:
+                    tracker.command_done_ok(res)
+                    continue
+                tracker.command_done_error(res.error.decode())
+                e = Exception("cmd %s error %s status %s", cmd, res.error, res.status)
+                seen_exc.append(e)
+                return seen_exc, result
+        return seen_exc, result
+
+    async def _reload_cmds(
```

Comment:
Looks very similar to _deploy_cmds(). Maybe reuse _reload_cmds inside _reload_cmds?